spring-xd-dirt/src/main/java/org/springframework/xd/dirt/plugins/spark/streaming/MessageBusReceiver.java

```diff
@@ -0,0 +1,116 @@
+
+import java.io.Serializable;
+import java.util.Properties;
+
+import org.apache.spark.storage.StorageLevel;
+import org.apache.spark.streaming.receiver.Receiver;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.integration.channel.DirectChannel;
+import org.springframework.messaging.Message;
+import org.springframework.xd.dirt.integration.bus.MessageBus;
+
+/**
+ * Spark {@link Receiver} implementation that binds to the MessageBus as a consumer.
+ * 
+ * @author Mark Fisher
+ * @author Ilayaperumal Gopinathan
+ */
+@SuppressWarnings({"rawtypes", "unchecked"})
+class MessageBusReceiver extends Receiver {
+
+	private static final long serialVersionUID = 1L;
+
+	/**
+	 * Logger.
+	 */
+	private static final Logger logger = LoggerFactory.getLogger(MessageBusReceiver.class);
+
+	private MessageBus messageBus;
+
+	private ConfigurableApplicationContext applicationContext;
+
+	private String channelName;
+
+	private final LocalMessageBusHolder messageBusHolder;
+
+	private final Properties messageBusProperties;
+
+	private final Properties moduleConsumerProperties;
+
+	public MessageBusReceiver(StorageLevel storageLevel, Properties messageBusProperties, Properties moduleConsumerProperties) {
+		this(null, storageLevel, messageBusProperties, moduleConsumerProperties);
+	}
+
+	public MessageBusReceiver(LocalMessageBusHolder messageBusHolder, StorageLevel storageLevel,
+			Properties messageBusProperties, Properties moduleConsumerProperties) {
+		super(storageLevel);
+		this.messageBusHolder = messageBusHolder;
+		this.messageBusProperties = messageBusProperties;
+		this.moduleConsumerProperties = moduleConsumerProperties;
+	}
+
+	public void setInputChannelName(String channelName) {
+		this.channelName = channelName;
+	}
+
+	@Override
+	public void onStart() {
+		logger.info("starting MessageBusReceiver");
+		MessageBus messageBus;
+		if (messageBusHolder != null) {
+			messageBus = messageBusHolder.get();
+		}
+		else {
+			applicationContext = MessageBusConfiguration.createApplicationContext(messageBusProperties);
+			messageBus = applicationContext.getBean(MessageBus.class);
+		}
+		messageBus.bindConsumer(channelName, new MessageStoringChannel(), moduleConsumerProperties);
+	}
+
+	@Override
+	public void onStop() {
+		logger.info("stopping MessageBusReceiver");
+		messageBus.unbindConsumers(channelName);
+		if (applicationContext != null) {
+			applicationContext.close();
+		}
+	}
+
+
+	/**
+	 * The {@link DirectChannel} that stores the received messages into Spark's memory.
+	 */
+	private class MessageStoringChannel extends DirectChannel implements Serializable {
+
+		private static final long serialVersionUID = 1L;
+
+		@Override
+		protected boolean doSend(Message<?> message, long timeout) {
+			//todo: probably just make this take the payload as is (especially if the inputType is applied as usual)...
```

Comment:
we need to remove this todo since we are now just sending the payload as is

we might want to create a JIRA issue for the configurable MessageConverter